plugin.video.youtube/resources/lib/youtube_plugin/kodion/plugin_runner.py

```diff
@@ -20,70 +27,115 @@
 __all__ = ('run',)
 
 _context = XbmcContext()
+_log = logging.getLogger(__name__)
 _plugin = XbmcPlugin()
 _provider = Provider()
-_profiler = Profiler(enabled=False, print_callees=False, num_lines=20)
+_profiler = Profiler(enabled=False,
+                     timer=Profiler.elapsed_timer,
+                     print_callees=False,
+                     num_lines=20)
 
 
 def run(context=_context,
+        log=_log,
         plugin=_plugin,
         provider=_provider,
         profiler=_profiler):
+    gc.disable()
```

Comment:
Could you make sure that there is a `try-finally` here, where in finally `gc.enable()` and `gc.collect()` is always called. Kodi re-uses Python instances and if your add-on fails, then the gc is disabled for that re-usable instance.